src/VisualStudio/VisualBasic/Impl/Options/VisualBasicSettingsManagerOptionSerializer.vb

```diff
@@ -118,6 +118,9 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.Options
                     Case FeatureOnOffOptions.FormatOnPaste.Name
                         Return Nothing
                 End Select
+            ElseIf key.Feature = ServiceFeatureOnOffOptions.OptionName Then
+                ' ClosedFileDiagnostics has been deprecated in favor of BasicClosedFileDiagnostics
```

Comment:
You should probably add a check for the key.Name as well in case we add some more keys to ServiceFeatureOnOffOptions in the future.